modin/pandas/series.py

```diff
@@ -15,7 +15,243 @@ def na_op():
     raise NotImplementedError("Not Yet implemented.")
 
 
-@_inherit_docstrings(pandas.Series, excluded=[pandas.Series, pandas.Series.__init__])
+class SeriesView(object):
+    """A wrapper class for pandas Series.
+
+    Note: The main use of this class is to help us implement inplace operations that
+        propagate their changes back to the DataFrame that a Series belongs to. We are
+        only need to use this object when `__getitem__` returns a pandas Series, or when
+        `loc`/`iloc` return a Series as well.
+
+    Important: This is not needed to replace every Series in Modin. For example, when an
+        operation on a Series returns a new Series, it does not need to return an object
+        of this class. It can return a Series because the new object does not have a
+        DataFrame that it is associated with.
+
+    """
+
+    def __init__(self, series=None, parent_df=None, loc=None):
+        self.series = series
```

Comment:
Can you add some asserts to make sure that all of the arguments are not none?

```suggestion
        self.series = series
        assert isinstance(parent_df) is modin.DataFrame
        assert loc != None
```